### PR TITLE
feat: Add DateUtils and DateService with tests

### LINKED ISSUE #264: Add DateUtils and delegate all date related tasks to this class
Add DateUtils and delegate all date-related tasks to this class from presenters and models

Taking this up

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #264 

- Add DateUtils and DateService
- Add tests
- Move all date related handling in DateService and DateUtils

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/utils/DateService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package org.fossasia.openevent.app.utils;
+
+import org.fossasia.openevent.app.data.models.Event;
+
+import java.text.ParseException;
+import java.util.Date;
+
+public class DateService {
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/utils/DateService.java]: Better we keep this (comparison and getting event status) in `Event.class` itself. That will be easier to manage as this class will become messy on further additions.

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/utils/DateService.java]: It's not true. Larger the class gets, the messier it is. See the composition of standard libraries in Java, almost all of them have just method names in their class and everything is delegated to specific objects.

https://github.com/ReactiveX/RxJava/blob/2.x/src/main/java/io/reactivex/Observable.java

And this is not something out of the blue I am advocating, this is Facade Design Pattern

https://en.wikipedia.org/wiki/Facade_pattern

Distributing code among classes is opposite of making things messy. As this is a model class, it already pretty huge, or else, there should be a simple rule, break the code if it exceeds 200 lines

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/utils/DateService.java]: >Distributing code among classes is opposite of making things messy

I was just suggesting to keep these two methods in event model as these are related to event model. No problem. :+1:

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/utils/DateService.java]: @arp95 @Shailesh351 @rohanagarwal94 Please review

**Pull request merged**